### PR TITLE
Panels: Allow interaction in corners when no title exists

### LINKED ISSUE #37153: Geomap: ZOOM-Control + cannot be used if no title is given
Grafana 8.1.0-29318pre
on Windows Server 2019

ZOOM-Control + cannot be used if no title is given
without a title, it should be moved down by the height of the title

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes https://github.com/grafana/grafana/issues/37153

When a panel does not have a title, the top band is still used to capture click events so we show the menu and can drag the panel around.  For some panels this is fine, but sometimes the top section has interactions that get blocked.

This PR proposes making the capture part smaller so at least the corners still work.  (the green is just to see what is happening)

header

header

### CODE REVIEW

File: public/sass/components/_panel_header.scss [hunk 1]
```diff
@@ -10,8 +10,10 @@ $panel-header-no-title-zindex: 1;
 .panel-container--no-title {
   .panel-header {
     position: absolute;
-    width: 100%;
+    width: calc(100% - 100px); // allow interaction in the corners
+    margin-left: 50px;
     z-index: $panel-header-z-index;
+    background: #0e78206e;
```
Comment (ryantxu) [inline public/sass/components/_panel_header.scss:16]: obviously will remove the background before commit :)

File: public/sass/components/_panel_header.scss [hunk 2]
```diff
@@ -10,8 +10,10 @@ $panel-header-no-title-zindex: 1;
 .panel-container--no-title {
   .panel-header {
     position: absolute;
-    width: 100%;
+    width: calc(100% - 100px); // allow interaction in the corners
+    margin-left: 50px;
```
Comment (leeoniya) [inline public/sass/components/_panel_header.scss:14]: i usually prefer `left`, though it does require `position: relative` or `position: absolute` (which is present here)

**Review by leeoniya — APPROVED** [review]

**Review by torkelo — APPROVED** [review]
Looks ok, but test with panel error or panel description before merging to make sure the panel description/error corner position is not affected. 

And long term I think: should improve [this](https://github.com/grafana/grafana/issues/31080), but not sure how moving a panel around is going to be handled in that design, so might have to have similar solution even with that 
https://github.com/grafana/grafana/issues/31080 should

**Comment by leeoniya** [comment]
obviously it takes the contents with it :(

i mean, we cannot leave these things in place there, cause the whole point is to clear that area, right?

image

**Comment by torkelo** [comment]
> obviously it takes the contents with it :(
> i mean, we cannot leave these things in place there, cause the whole point is to clear that area, right?

Dam, well then it's back to the drawing board :(

**Comment by torkelo** [comment]
Or continue work on the new PanelChrome, https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx , where I think a no title option we can place the move handle in the absolutely positioned hover/focus menu. 

This does highlight an issue with that new design, it will now support the current panel description info icon & the no header (absolutely positioned) mode. Think long term we have to introuce a "Hide panel header" option instead of a "empty title" = no header. Hard to express everything here but panel chrome (panel header, menu, etc) is due to a major redesign. 

But what to do short term, temp fix for 8.1, can we move the zoom controls down a bit?

**Comment by leeoniya** [comment]
> But what to do short term, temp fix for 8.1, can we move the zoom controls down a bit?

i can imagine having visible map markers at the top that will be obstructed by any transparent panel header and cause similar UX problems, and map markers cannot be moved, though i guess you can pan the map when you discover this easter egg :)

**Comment by leeoniya** [comment]
some inspiration:

image

**Comment by ryantxu** [comment]
@leeoniya weird -- this is what I see with an error, how did you get yours?

image

**Comment by ryantxu** [comment]
Talked with @leeoniya -- his example was using devtools, not the PR.  This is with the PR:
image

So error is OK (but no interaction), Info has a corner -- Perhaps with info we can avoid `panel-container--no-title` like we do with error :thinking:

**Comment by ryantxu** [comment]
2/3.... lets see:
image

File: public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx [hunk 3]
```diff
@@ -173,12 +173,7 @@ export class PanelChromeAngularUnconnected extends PureComponent<Props, State> {
 
   hasOverlayHeader() {
     const { panel } = this.props;
-    const { errorMessage, data } = this.state;
```
Comment (ryantxu) [inline public/app/features/dashboard/dashgrid/PanelChromeAngular.tsx:176]: @torkelo if we move the corner out of the title element, we don't need this check

File: public/app/features/dashboard/dashgrid/PanelHeader/PanelHeader.tsx [hunk 4]
```diff
@@ -35,15 +35,15 @@ export const PanelHeader: FC<Props> = ({ panel, error, isViewing, isEditing, dat
   return (
     <>
       <PanelHeaderLoadingIndicator state={data.state} onClick={onCancelQuery} />
+      <PanelHeaderCorner
```
Comment (ryantxu) [inline public/app/features/dashboard/dashgrid/PanelHeader/PanelHeader.tsx:38]: @torkelo any reason this is a bad idea?

**Comment by torkelo** [comment]
Wont this also cause issues with small panels where the title cannot extend the full width of the panel?

**Comment by leeoniya** [comment]
we can go with something like `left: min(100px, 3%)` to make sure it doesn't shrink too much.

**Comment by ryantxu** [comment]
> Wont this also cause issues with small panels where the title cannot extend the full width of the panel?

With long titles, this is not an issue because there is explicit space for them!  however you are right that with very small panels, the absolute sizes would get so small you could never open the editor.  I added @leeoniya's min() suggestion and now have:
resize

**Pull request merged**